### PR TITLE
[Front] Refacto Incident component into .tsx pure function 

### PR DESCRIPTION
**Pull request by marieflorescontact**
### Proposed changes

* Transform Incidents components into .tsx pure function 

### Checklist

- [x] I consider the submitted work as finished

``

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentEditionOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+    incident_type: incident.incident_type,
+    severity: incident.severity,
+    createdBy,
+    objectMarking,
+    objectAssignee,
+    x_opencti_workflow_id: status,
+    status: incident.status,
+    workflowEnabled: incident.workflowEnabled,
+    is_inferred: isInferred,
+    names: [
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentEditionOverview.tsx]: This property is used ?

File: opencti-platform/opencti-front/src/private/components/events/incidents/Root.tsx [hunk 2]
```diff
@@ -209,21 +192,23 @@ class RootIncident extends Component {
...
-  children: PropTypes.node,
-  match: PropTypes.object,
+  );
 };
 
-export default withRouter(RootIncident);
+const RootIncident = () => {
+  const { incidentId } = useParams() as { incidentId: string };
+  const queryRef = useQueryLoading < RootIncidentQuery >(incidentQuery, { id: incidentId });
+  return queryRef ? (
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/events/incidents/Root.tsx]: Do you have any blink on the topbar when you switch right menu in the knowledge page ?

A few days ago, we fix a bug on blinking. You can see the fix in the regions/Root.tsx file for example (the topbar and the StixCoreObjectKnowledgeBar are not in the same place anymore)

File: opencti-platform/opencti-front/src/private/components/events/Incidents.tsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+import { Filters } from '../../../components/list_lines';
+import { LOCAL_STORAGE_KEY_CASE } from '../cases/Incidents';
+import { IncidentLineDummy } from './incidents/IncidentLine';
+import { IncidentCardDummy } from './incidents/IncidentCard';
+import { IncidentsCardsPaginationQuery } from './incidents/__generated__/IncidentsCardsPaginationQuery.graphql';
+
+const Incidents: FunctionComponent = () => {
+  const { helper } = useContext(UserContext);
+  const { viewStorage, helpers, paginationOptions } = usePaginationLocalStorage<IncidentsLinesPaginationQuery$variables>(
+    LOCAL_STORAGE_KEY_CASE,
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/events/Incidents.tsx]: events/incidents and cases/incidents are not the same object types ! you should redefine a key for Incidents.

File: opencti-platform/opencti-front/src/private/components/events/Incidents.tsx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  const {
+    sortBy,
+    orderAsc,
+    searchTerm,
+    filters,
+    openExports,
+    numberOfElements,
+  } = viewStorage;
+
+  const [view, saveView] = useViewStorage('view-incidents');
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/events/Incidents.tsx]: 'view-incidents' = LOCAL_STORAGE_KEY

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentKnowledge.tsx [hunk 5]
```diff
@@ -13,29 +14,46 @@ import StixDomainObjectHeader from '../../common/stix_domain_objects/StixDomainO
...
+        entityType={'Incident'}
+        stixDomainObject={incident}
+        PopoverComponent={IncidentPopover }
+      />
+      <Switch>
         <Route
           exact
           path="/dashboard/events/incidents/:incidentId/knowledge/relations/:relationId"
-          render={(routeProps) => (
+          render={(routeProps: any) => (
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentKnowledge.tsx]: you shouldn't use 'any' as a type

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentKnowledge.tsx [hunk 6]
```diff
@@ -46,7 +64,7 @@ class IncidentKnowledgeComponent extends Component {
         <Route
           exact
           path="/dashboard/events/incidents/:incidentId/knowledge/sightings/:sightingId"
-          render={(routeProps) => (
+          render={(routeProps: any) => (
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentKnowledge.tsx]: same here and in other lines

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentLine.tsx [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+            </div>
+            <div
+              className={classes.bodyItem}
+              style={{ width: dataColumns.incident_type.width }}
+            >
+              <Chip
+                classes={{ root: classes.chipInList }}
+                color="primary"
+                variant="outlined"
+                label={data.incident_type || t('Unknown')}
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentLine.tsx:137]: translation of data.incident_type t('data.incident_type')

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentLine.tsx [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+                variant="inList"
+              />
+            </div>
+            <div
+              className={classes.bodyItem}
+              style={{ width: dataColumns.objectLabel.width }}
+            >
+              <StixCoreObjectLabels
+                variant="inList"
+                labels={data.objectLabel}
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentLine.tsx:156]: translation to add ?

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentLine.tsx [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+        }
+      />
+      <ListItemIcon classes={{ root: classes.goIcon }}>
+        <KeyboardArrowRight />
+      </ListItemIcon>
+    </ListItem>
+  );
+};
+
+export const IncidentLineDummy = ({ dataColumns }: { dataColumns: DataColumns; }) => {
```
Comment (Archidoit) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentLine.tsx]: not necessary semi-colon

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentCard.js [hunk 10]
```diff
@@ -1,259 +0,0 @@
...
-  withStyles(styles),
-)(IncidentCardFragment);
-
-class IncidentCardDummyComponent extends Component {
-  render() {
-    const { classes } = this.props;
-    return (
-      <Card classes={{ root: classes.cardDummy }} variant="outlined">
-        <CardActionArea classes={{ root: classes.area }}>
-          <CardHeader
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentCard.js:190]: Why the IncidentCardDummyComponent in the new file incidentCard.tsx is not the same as here ?

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentDetails.tsx [hunk 11]
```diff
@@ -137,6 +171,8 @@ const IncidentDetailsComponent = ({ incident }) => {
               parameters={{ title: t('Entities distribution') }}
               variant="inEntity"
               height={300}
+              startDate={incident.first_seen}
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentDetails.tsx:174]: Why are you adding this ?

**Comment by RomuDeuxfois** [comment]
Bugs found : 
- In card mode, when we sort by name, the card are not sorting
- In edition the confidence field is not filled
- The assignee, severity and incident type fields are not translate
- Modified with the enforce reference setting is not working

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentDetails.tsx (hunk 11 above)
Comment (marieflorescontact) [reply opencti-platform/opencti-front/src/private/components/events/incidents/IncidentDetails.tsx:174]: The [`StixCoreObjectsDonut`](https://github.com/OpenCTI-Platform/opencti/blob/master/opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectsDonut.js) component needs `startDate` and `endDate`

**Comment by marieflorescontact** [comment]
> Bugs found :
> 
> * In card mode, when we sort by name, the card are not sorting `fixed`
> * In edition the confidence field is not filled `fixed`
> * The assignee, severity and incident type fields are not translate `fixed`
> * Modified with the enforce reference setting is not working  `fixed`

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**